File: dna/dna.rb
#!/usr/bin/env ruby

index = {'A' => 0, 'C' => 1, 'G' => 2, 'T' => 3}
count = [0, 0, 0, 0]
ARGF.each do |line|
  line.chomp!
  line.each_char do|c|
      count[index[c]] += 1
  end
end
puts count.join(' ')

File: dna.pl
#!/usr/bin/env perl

use strict;
use warnings;

my %index = (A => 0, C => 1, G => 2, T => 3);
my @count = (0, 0, 0, 0);
while (my $input = <STDIN>) {
    chomp $input;
    map { $count[$index{$_}]++ } split(//, $input);
}
print join(' ', @count), "\n";
